cirq/circuits/circuit_dag.py

```diff
@@ -209,3 +209,24 @@ def findall_nodes_until_blocked(
                 remaining_dag.remove_node(node)
                 continue
             yield node
+
+    def findall_nodes_from_set(
+        self, input_set: Iterable[Unique[ops.Operation]]
+    ) -> Iterator[ops.Operation]:
+        """Finds all nodes from the given set.
+
+        Args:
+            input: The input set of Operation nodes.
+        """
+        for node in self.ordered_nodes():
+            if node in input_set:
+                yield node.val
+
+    def factorize(self) -> Iterator['cirq.CircuitDag']:
+        """Tries to factorize the underlying graph, using connected components algorithm.
+        If no factorization is possible, returns a sequence with a single element (itself).
+        """
+        for c in networkx.weakly_connected_components(self):
+            yield CircuitDag.from_ops(
+                self.findall_nodes_from_set(c), can_reorder=self.can_reorder, device=self.device
```

Comment:
What impact does `can_reorder` and `device` have on the output of `factorize`?

Reply:
How do you get a normal `Circuit` back? Should we return `Circuit`s instead of `CircuitDag`s in this function?

Reply:
can_reorder is a method to decide whether two nodes (e.g. operations) are mutually independent. The default implementation checks that the qubits which participate in these operations don't intersect, but it can be possible to supply a different method at CircuitDag construction.

The device is there to properly reconstruct the original circuit, it isn't used within CircuitDag itself.